### PR TITLE
Moniika moniika pinch zoom config

### LINKED ISSUE #2782: Pinch to zoom can't be configured / turned off
**Describe the bug**

Pinch to zoom doesn't respect zoom options in order to figure out whether or not to zoom the workspace. 

**To Reproduce**

1. Configure the playground without any zoom options.
2. Browse to the playground on a touch device. 
3. Try to pinch to zoom the workspace
4. It works, when it shouldn't..

**Expected behavior**

I expect a developer can turn off pinch to zoom with the zoom options.

**Smartphone (please complete the following information):**
 - Device: Iphone X
 - OS: IOS 12
 - Browser: Safari

**Additional context**

Currently there is no zoom option for pinch to zoom. We have a few options: 
- We could overload wheel. 
- We could create a new option.
- We could just check for the existence of the zoom option (ie: any zoom option, enables it). 

I feel if we create a zoom option for it now, we'd have to communicate to people that they have to now opt into it through that option.

### PR DESCRIPTION
**Pull request by moniika**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

#2782

### Proposed Changes

Adds pinch zoom option.

### Reason for Changes

Allows for pinch zoom to be configured.

### Test Coverage

Tested on:
* Desktop Chrome

### CODE REVIEW

File: core/workspace_svg.js [hunk 1]
```diff
@@ -1799,12 +1801,6 @@ Blockly.WorkspaceSvg.prototype.setBrowserFocus = function() {
  *     amount values zoom in.
  */
 Blockly.WorkspaceSvg.prototype.zoom = function(x, y, amount) {
-  // TODO (#2782): Consider removing once pinch understands zoom configuration
```
Comment (rachel-fenichel) [inline core/workspace_svg.js]: Did you confirm that nothing breaks in mutators or the flyout?

File: core/touch_gesture.js [hunk 2]
```diff
@@ -257,28 +257,33 @@ Blockly.TouchGesture.prototype.handleTouchMove = function(e) {
...
-  var pointers = Object.keys(this.cachedPoints_);
-  // If two pointers are down, check for pinch gestures
-  if (pointers.length == 2) {
-    // Calculate the distance between the two pointers
-    var point0 = this.cachedPoints_[pointers[0]];
-    var point1 = this.cachedPoints_[pointers[1]];
-    var moveDistance = Blockly.utils.Coordinate.distance(point0, point1);
-    var startDistance = this.startDistance_;
-    var scale = this.touchScale_ = moveDistance / startDistance;
+  var workspace = this.startWorkspace_;
```
Comment (rachel-fenichel) [inline core/touch_gesture.js]: Pull this out into another function (`handlePinch`?)

File: core/touch_gesture.js [hunk 3]
```diff
@@ -257,28 +257,33 @@ Blockly.TouchGesture.prototype.handleTouchMove = function(e) {
...
+  if (workspace.options.zoomOptions && workspace.options.zoomOptions.pinch) {
+    var pointers = Object.keys(this.cachedPoints_);
+    // If two pointers are down, check for pinch gestures
+    if (pointers.length == 2) {
+      // Calculate the distance between the two pointers
+      var point0 = this.cachedPoints_[pointers[0]];
+      var point1 = this.cachedPoints_[pointers[1]];
+      var moveDistance = Blockly.utils.Coordinate.distance(point0, point1);
+      var startDistance = this.startDistance_;
+      var scale = this.touchScale_ = moveDistance / startDistance;
```
Comment (rachel-fenichel) [inline core/touch_gesture.js]: I know this already existed, but I'm not a fan of chaining assignment like this.  I prefer just setting `this.touchScale_` and using it in place of `scale` in the two locations below.

**Comment by RoboErikG** [comment]
@moniika please follow up.

**Pull request closed without merging**